stablehlo/reference/Ops.cpp

```diff
@@ -133,6 +134,345 @@ Tensor evalConvertOp(const Tensor &operand, TensorType resultType) {
   return result;
 }
 
+Tensor evalConvolutionOp(
+    const Tensor &lhs, const Tensor &rhs, ArrayRef<int64_t> windowStrides,
+    SmallVector<SmallVector<int64_t>> padding, ArrayRef<int64_t> lhsDilation,
+    ArrayRef<int64_t> rhsDilation, ArrayRef<bool> windowReversal,
+    Axis inputBatchDimension, Axis inputFeatureDimension,
+    Axes inputSpatialDimensions, Axis kernelInputFeatureDimension,
+    Axis kernelOutputFeatureDimension, Axes kernelSpatialDimensions,
+    Axis outputBatchDimension, Axis outputFeatureDimension,
+    Axes outputSpatialDimensions, int64_t featureGroupCount,
+    int64_t batchGroupCount, TensorType resultType) {
+  Tensor result(resultType);
+
+  if (resultType.getNumElements() == 0) return result;
```

Comment:
This one in particular is needed because we are not looping on the result tensor but instead creating our own index. When one of the tensors' dimension size is zero, it will still try to create an index iterator out of it, which will cause issues.

Reply:
I'll put a comment on that to explain its necessity.